idx/cassandra/cassandra.go

```diff
@@ -122,35 +68,38 @@ type cqlIterator interface {
 	Close() error
 }
 
-func New() *CasIdx {
-	cluster := gocql.NewCluster(strings.Split(hosts, ",")...)
-	cluster.Consistency = gocql.ParseConsistency(consistency)
-	cluster.Timeout = timeout
+func New(cfg *IdxConfig) *CasIdx {
+	if err := cfg.Validate(); err != nil {
+		log.Fatalf("cassandra-idx: %s", err)
+	}
+	cluster := gocql.NewCluster(strings.Split(cfg.hosts, ",")...)
+	cluster.Consistency = gocql.ParseConsistency(cfg.consistency)
+	cluster.Timeout = cfg.timeout
 	cluster.ConnectTimeout = cluster.Timeout
-	cluster.NumConns = numConns
-	cluster.ProtoVersion = protoVer
-	cluster.DisableInitialHostLookup = disableInitialHostLookup
-	if ssl {
+	cluster.NumConns = cfg.numConns
+	cluster.ProtoVersion = cfg.protoVer
+	cluster.DisableInitialHostLookup = cfg.disableInitialHostLookup
+	if cfg.ssl {
 		cluster.SslOpts = &gocql.SslOptions{
-			CaPath:                 capath,
-			EnableHostVerification: hostverification,
+			CaPath:                 cfg.capath,
+			EnableHostVerification: cfg.hostverification,
 		}
 	}
-	if auth {
+	if cfg.auth {
 		cluster.Authenticator = gocql.PasswordAuthenticator{
-			Username: username,
-			Password: password,
+			Username: cfg.username,
+			Password: cfg.password,
 		}
 	}
 
 	idx := &CasIdx{
 		MemoryIdx: *memory.New(),
+		cfg:       cfg,
 		cluster:   cluster,
 	}
-	if updateCassIdx {
-		idx.writeQueue = make(chan writeReq, writeQueueSize)
+	if cfg.updateCassIdx {
+		idx.writeQueue = make(chan writeReq, cfg.writeQueueSize)
 	}
-	updateInterval32 = uint32(updateInterval.Nanoseconds() / int64(time.Second))
```

Comment:
why did we remove this here and move it to IdxConfig.Validate()
validation should validate the config. setting this property must not be forgotten, so it should not be in the Validate method (which is optional)

Reply:
`Validate` is not optional, it is called at the beginning of `New`. This way `updateInterval` is validated before it is attempted to be used in this assignment.

Reply:
ok but it smells like bad code hygiene and for someone reading the code, that logic is not obvious.
and the question remains, why is this line in validate if it has nothing to do with validation, and why is it removed from New() which seems like a better place to put it.

Reply:
Fair enough, I suppose it really doesn't have anything to do with validation itself, other than relying on validation to run before it is assigned. I will move it back to New() and adjust the comments.

Reply:
On phone now but if that property is a derivative of the cfg that's set in New it should probably be a field of the index struct not of the config object